File: app/src/main/java/br/pedroso/movies/shared/data/retrofit/entities/RetrofitMovieEntity.java
package br.pedroso.movies.shared.data.retrofit.entities;

public class RetrofitMovieEntity {
    private Integer id;
    private String poster_path;
    private String title;
    public Double vote_average;
    public String release_date;
    public String overview;

    public RetrofitMovieEntity(Integer id, String poster_path, String title, Double vote_average, String release_date, String overview) {
        this.id = id;
        this.poster_path = poster_path;
        this.title = title;
        this.vote_average = vote_average;
        this.release_date = release_date;
        this.overview = overview;
    }

    public Integer getId() {
        return id;
    }

    public String getPosterPath() {
        return poster_path;
    }

    public String getTitle() {
        return title;
    }

    public Double getVoteAverage() {
        return vote_average;
    }

    public String getOverview() {
        return overview;
    }

    public String getReleaseDate() {
        return release_date;
    }
}

File: app/src/main/kotlin/br/pedroso/upcomingmovies/network/entities/RetrofitMovieEntity.kt
package br.pedroso.upcomingmovies.network.entities

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
class RetrofitMovieEntity(
    @SerialName("id") val id: Int,
    @SerialName("poster_path") val posterPath: String = "",
    @SerialName("title") val title: String,
    @SerialName("vote_average") var voteAverage: Double,
    @SerialName("release_date") var releaseDate: String,
    @SerialName("overview") var overview: String,
)
